File: features/discovery/filter/src/main/java/com/tokopedia/filter/newdynamicfilter/database/FilterDatabaseClient.java
package com.tokopedia.filter.newdynamicfilter.database;

import android.content.Context;
import androidx.room.Room;

public class FilterDatabaseClient {
    private static FilterDatabaseClient mInstance;

    private FilterDatabase filterDatabase;

    private FilterDatabaseClient(Context context) {
        filterDatabase = Room.databaseBuilder(context.getApplicationContext(),
                FilterDatabase.class, "Filter.db").fallbackToDestructiveMigration().build();
    }

    public static synchronized FilterDatabaseClient getInstance(Context context) {
        if (mInstance == null) {
            mInstance = new FilterDatabaseClient(context);
        }
        return mInstance;
    }

    public FilterDatabase getFilterDatabase() {
        return filterDatabase;
    }
}

File: features/discovery/filter/src/main/java/com/tokopedia/filter/newdynamicfilter/database/FilterDatabaseClient.kt
package com.tokopedia.filter.newdynamicfilter.database

import android.content.Context
import androidx.room.Room

class FilterDatabaseClient private constructor(context: Context) {

    val filterDatabase: FilterDatabase = Room.databaseBuilder(context.applicationContext,
            FilterDatabase::class.java, "Filter.db").fallbackToDestructiveMigration().build()

    companion object {
        private var mInstance: FilterDatabaseClient? = null

        @Synchronized
        fun getInstance(context: Context): FilterDatabaseClient {
            if (mInstance == null) {
                mInstance = FilterDatabaseClient(context)
            }
            return mInstance as FilterDatabaseClient
        }
    }
}
